File: Semester 1/Class02_Computer_Contest_Lv_2_Notes_and_Homework_Sep_18/lvl2_hw_sol/lvl2_modern_art.py
import sys
from collections import deque
input = sys.stdin.readline
m, n, k = int(input()), int(input()), int(input())
row, col = [0]*(m+1), [0]*(n+1)
for _ in range(k):
    op, x = input().split()
    if op == 'R':
        row[int(x)]+=1
    else:
        col[int(x)]+=1
evenr, oddr, evenc, oddc = 0, 0, 0, 0
for i in range(1, m+1):
    if row[i] % 2 == 0:
        evenr+=1
    else:
        oddr+=1
for i in range(1, n+1):
    if col[i] % 2 == 0:
        evenc += 1
    else:
        oddc += 1
print(evenr * oddc + oddr * evenc)

File: Semester 1/Class02_Computer_Contest_Lv_2_Notes_and_Homework_Sep_18/lvl2_hw_sol/lvl2_modern_art.java
import java.util.*;
public class test2 {
	public static void main(String [] args) {
		Scanner in = new Scanner(System.in);
		int m = in.nextInt(), n = in.nextInt(), k = in.nextInt();
		int [] row = new int[m+1], col = new int[n+1];
		for(int i=1; i<=k; i++) {
			char op = in.next().charAt(0); int x = in.nextInt();
			if(op == 'R') row[x]++;
			else col[x]++;
		}
		int evenr = 0, oddr = 0, evenc = 0, oddc = 0;
		for(int i=1; i<=m; i++) {
			if(row[i] % 2 == 0) evenr ++;
			else oddr++;
		}
		for(int i=1; i<=n; i++) {
			if(col[i] % 2 == 0) evenc++;
			else oddc++;
		}
		System.out.println(evenr * oddc + oddr * evenc);
	}
}